aiida_quantumespresso/parsers/pp.py

```diff
@@ -0,0 +1,348 @@
+        9: 'e/bohr^3', # Charge density minus superposition of atomic densities
+        10: 'states/bohr^3', # Integrated local density of states (ILDOS)
+        11: 'Ry', # Bare + Hartree potential
+        12: 'Ry', # the sawtooth electric field potential
+        13: 'mu_B', # Noncollinear magnetisation, Bohr magnetons
+        17: 'e/bohr^3', # All electron charge density
+        18: 'T', # The exchange and correlation magnetic field in the noncollinear case
+        19: '1', # Reduced density gradient - see dx.doi.org/10.1021/ct100641a, Eq.1 - dimensionless
+        20: 'e/bohr^5', # Product of the electron density and the second eigenvalue of the electron-density Hessian matrix, see: dx.doi.org/10.1021/ct100641a, with sign of second eigenvalue
+        21: 'e/bohr^3', # All electron charge density, PAW case
+        22: 'Ry/bohr^3', # Kinetic energy density
+    }
+
+    def parse(self, **kwargs):
+        """
+        Parse raw files retrieved from remote dir
+        """
+
+        # A retrieved folded is required
+        try:
+            self.retrieved
+        except exceptions.NotExistent:
+            return self.exit_codes.ERROR_NO_RETRIEVED_FOLDER
+
+        # The stdout is required for parsing
+        filename_stdout = self.node.get_attribute('output_filename')
+        if filename_stdout not in self.retrieved.list_object_names():
+            return self.exit_codes.ERROR_OUTPUT_STDOUT_MISSING
+
+        try:
+            stdout_raw = self.retrieved.get_object_content(filename_stdout)
+        except (IOError, OSError):
+            return self.exit_codes.ERROR_OUTPUT_STDOUT_READ
+
+        # The post-processed data should have been written to file
+        filename_data = PpCalculation._FILEOUT
+        if filename_data not in self.retrieved.list_object_names():
+            return self.exit_codes.ERROR_OUTPUT_DATAFILE_MISSING
+
+        try:
+            data_raw = self.retrieved.get_object_content(filename_data)
+        except (IOError, OSError):
+            return self.exit_codes.ERROR_DATAFILE_READ
+
+        # Parse stdout
+        try:
+            logs, self.output_parameters = self.parse_stdout(stdout_raw)
+        except Exception:
+            self.logger.error(traceback.format_exc())
+            return self.exit_codes.ERROR_UNEXPECTED_PARSER_EXCEPTION
+
+        # Print the logs
+        self.emit_logs(logs)
+
+        # Scan logs for known errors
+        if 'ERROR_PARENT_XML_MISSING' in logs['error']:
+            return self.exit_codes.ERROR_PARENT_XML_MISSING
+        if 'ERROR_OUTPUT_STDOUT_INCOMPLETE' in logs['error']:
+            return self.exit_codes.ERROR_OUTPUT_STDOUT_INCOMPLETE
+
+        # Parse the post-processed-data according to what kind of data file was produced
+        if self.output_parameters['output_format'] == 'gnuplot':
+            if self.output_parameters['plot_type'] == '2D polar on a sphere':
+                parsed_data = self.parse_gnuplot_polar(data_raw)
```

Comment:
I am wondering now if we should maybe move the data file to the `retrieve_temporary_list`. The reason is that these files can be quite big (correct?) and we are parsing it essentially in its entirety into an `ArrayData`. In a sense we are then duplicating the content, because the original raw file is stored in the file repo of the calculation node as well as in a parsed version in the `ArrayData` output node. Do we really need the original raw file if we are storing it as a node as well? If not, then we can retrieve this file in the temporary retrieved folder, which still allows to parse it, but the engine will clean it after parsing and not store it in the repo. Maybe this is too much work for now and if you think it makes sense we can simply open an issue for this. @greschd what are your thoughts?

Reply:
Hmm, I'm not sure about this. For my own use it would definitely be fine to just discard it. However, I think the main reason `pp.x` even supports different output formats is so that they can effortlessly be fed into different plotting tools. If we discard the file, the onus is on us to provide compatibility with these tools (from the `ArrayData`).

Maybe a good solution would be to discard the file by default, and provide a setting to keep it?

Reply:
I think the discarding by default and providing a setting to override would be the best solution. But we can do this in a separate PR so that we can merge this one soon

Reply:
Shouldn't we change this PR to _always_ discard then, so that the change of adding a setting is backwards compatible?

Reply:
These can be GBs easily for a dense grid in a big supercell.  I'd agree that if we parse, we don't need the original. This is loosely analogous to the argument over what to do with MD trajectories.

Reply:
So as it stands, when one asks pp.x to write out to file in particular format, pp.x produces two files: 1. the 3D gridded quantity in a custom format, and 2. the post-processed data converted into a particular format, and reduced to the dimension requested by `iflag`. 
In the most recent implementation, we never use the first file, and so there is no need to retrieve it. The second, the file in a format we choose according to what we can parse, we temporarily retrieve, parse and discard.